scipy/stats/stats.py

```diff
@@ -5476,6 +5489,20 @@ def combine_pvalues(pvalues, method='fisher', weights=None):
         Xsq = -2 * np.sum(np.log(pvalues))
         pval = distributions.chi2.sf(Xsq, 2 * len(pvalues))
         return (Xsq, pval)
```

Comment:
Maybe it is worth to compute statistic, pval for each method and then have one return statement at the end

Reply:
Yes, that was the Idea but not critical. The name `Xsq` is not a good fit for all the methods though.

Reply:
(just as a bystander)
I don't like `x` or `X` for `chi`.   (I read x as eks and not as chi)
We have one of those in statsmodels and I never remember that `x2` is supposed to be `chi2` (i.e. pearson chisquared in that case).